src/simulation.cpp

```diff
@@ -428,6 +428,18 @@ void finalize_batch()
     auto filename = settings::path_output + "surface_source";
     auto surf_work_index =
       mpi::calculate_parallel_index_vector(simulation::surf_source_bank.size());
+
+    // Sort, because the file ordering will differ based on the order that
+    // threads finish work in. We want to produce the same output every time
+    // given the same random seed. We probably don't need to sort globally over
+    // MPI ranks to ensure reproducible output.
+    std::sort(simulation::surf_source_bank.begin(),
+      simulation::surf_source_bank.end(),
+      [](const SourceSite& l, const SourceSite& r) {
+        return std::tie(l.parent_id, l.progeny_id) <
+               std::tie(r.parent_id, r.progeny_id);
+      });
```

Comment:
This was added with the help of @gridley (to be more precise, he wrote this part of the code! :smiley:) - see https://github.com/openmc-dev/openmc/pull/2500. We thought this would solve the *surface source bug*. Since you solved this one, I will remove this block again.